### PR TITLE
Remove `FrameStableClock` (and redirect usages to `FrameStabilityContainer`)

### PR DESCRIPTION
**Pull request by peppy**
- [x] Depends on #19775.

### CODE REVIEW

File: osu.Game.Tests/Visual/Gameplay/TestSceneGameplaySamplePlayback.cs [hunk 1]
```diff
@@ -1,8 +1,6 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
-#nullable disable
```
Comment (bdach) [inline osu.Game.Tests/Visual/Gameplay/TestSceneGameplaySamplePlayback.cs:4]: Enabling NRT in this file seems a bit drive-by (there seem to be no actual changes in this file related to `FrameStableClock`) but ok.

File: osu.Game/Rulesets/UI/FrameStabilityContainer.cs [hunk 2]
```diff
@@ -63,20 +67,17 @@ public FrameStabilityContainer(double gameplayStartTime = double.MinValue)
         /// </remarks>
         private int direction = 1;
 
-        [BackgroundDependencyLoader(true)]
-        private void load(IGameplayClock clock)
+        [BackgroundDependencyLoader]
+        private void load(IGameplayClock? gameplayClock)
```
Comment (bdach) [inline osu.Game/Rulesets/UI/FrameStabilityContainer.cs:71]: Are we still changing `[BackgroundDependencyLoader(true)]` to NRT-annotated params after [the latest mono-related fiasco](https://discord.com/channels/188630481301012481/188630652340404224/1006802093232881715)? Not that I imagine this particular one will cause issues, but it's a bit touch-and-go and I'm not sure where we're at regarding this.

**Review by bdach — COMMENTED** [review]
Looks mostly fine. Scariest thing is, I'm not really sure I can tell if the clock shuffling is correct. It *looks* okay, but I'm *not actually sure still*.

File: osu.Game/Rulesets/UI/FrameStabilityContainer.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Rulesets/UI/FrameStabilityContainer.cs:71]: I did consider that for this specific usage, but I do believe it's only a problem in tests. Can revert if preferred.

File: osu.Game.Tests/Visual/Gameplay/TestSceneGameplaySamplePlayback.cs (hunk 1 above)
Comment (peppy) [reply osu.Game.Tests/Visual/Gameplay/TestSceneGameplaySamplePlayback.cs:4]: I think I originally had changed in this file...

**Review by smoogipoo — APPROVED** [review]

File: osu.Game/Rulesets/UI/FrameStabilityContainer.cs (hunk 2 above)
Comment (smoogipoo) [reply osu.Game/Rulesets/UI/FrameStabilityContainer.cs:71]: Should be okay. Honestly would like to see NET6 prioritised still...
